### PR TITLE
Merge links into nodes: update pf and sc tests 

### PR DESCRIPTION
**Pull request by Jerry-Jinfeng-Guo**
This PR provides tests that are updated according to new links calculation results. Some differ from https://github.com/PowerGridModel/power-grid-model/pull/1507 due to calculation by hand. 

Short circuit calculation test case for line-to-itself edge case is updated. To be followed-up in https://github.com/PowerGridModel/power-grid-model/pull/1566.

### CODE REVIEW

File: tests/data/power_flow/dummy-test-line-into-itself/asym_output_batch.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
...
         {"id": 2, "energized": 1, "u_pu": [1.0489305412180272, 1.0488408010290586, 1.047862975371989], "u": [6056.003303334478, 6055.4851881118975, 6049.8397090486005], "u_angle": [-0.0031192166967468683, -2.0971289939257418, 2.0904277197067258], "p": [0.0016991138504116397, 0.0037284466839334995, -0.003119483334405209], "q": [-0.0026769212471873154, -0.0064291658102272617, -0.0016728438498193419]},
-        {"id": 3, "energized": 1, "u_pu": [1.0489305377088716, 1.0488407990932342, 1.0478629702944795], "u": [6056.0032830743594, 6055.48517693541, 6049.8396797335854], "u_angle": [-0.0031192079794357627, -2.0971289867084129, 2.0904277299238556], "p": [-440102.09839980368, -330020.10079295578, -549008.40602486604], "q": [-220051.05491385818, -220013.40255902623, -219603.34729454646]}
+        {"id": 3, "energized": 1, "u_pu": [1.0489305411522316, 1.0488408010053642, 1.04786297529498795], "u": [6056.0033029546424, 6055.485187975122, 6049.8397086040414], "u_angle": [-0.0031192079794357627, -2.0971289867084129, 2.0904277299238556], "p": [-440102.09839980368, -330020.10079295578, -549008.40602486604], "q": [-220051.05491385818, -220013.40255902623, -219603.34729454646]}
```
Comment (nitbharambe) [inline tests/data/power_flow/dummy-test-line-into-itself/asym_output_batch.json:11]: Why modify pf tests?

File: tests/data/short_circuit/dummy-test-line-into-itself/params.json [hunk 2]
```diff
@@ -5,5 +5,9 @@
     "default": 1e-9,
     "i_(.+)?": 1e-5
   },
-  "short_circuit_voltage_scaling": "maximum"
+  "short_circuit_voltage_scaling": "maximum",
+  "xfail": {
+    "raises": "AssertionError",
+    "reason": "Bug to be fixed in https://github.com/PowerGridModel/power-grid-model/pull/1566"
+  }
```
Comment (nitbharambe) [inline tests/data/short_circuit/dummy-test-line-into-itself/params.json]: Not all of it would be fixed by #1566. So the independent problem of 1566 needs to be accounted first and then short circuit tests needs to be modified for merge with links behaviour.

File: tests/data/power_flow/handling-of-links/merge-nodes-by-links-1-line-with-line-to-self/sym_output.json [hunk 3]
```diff
@@ -98,13 +98,13 @@
                 "energized": 1,
                 "loading": 0.0,
                 "p_from": 21084319.029628307,
-                "q_from": 5195613.111142274,
-                "i_from": 1195.2496789651816,
-                "s_from": 21715038.672399823,
+                "q_from": 5541259.927274941,
```
Comment (nitbharambe) [inline tests/data/power_flow/handling-of-links/merge-nodes-by-links-1-line-with-line-to-self/sym_output.json:101]: These needs to be addressed

Comment (nitbharambe) [reply tests/data/power_flow/handling-of-links/merge-nodes-by-links-1-line-with-line-to-self/sym_output.json:101]: @Jerry-Jinfeng-Guo Confirmed this change is okay and that this test is indeed incorrect in main.

File: tests/data/power_flow/handling-of-links/merge-nodes-by-links-1-line-with-line-to-self/sym_output.json
Comment (nitbharambe) [inline tests/data/power_flow/handling-of-links/merge-nodes-by-links-1-line-with-line-to-self/sym_output.json:66]: The below values of link need to be equal to these values flipped. Hence the change in this PR is correct.

File: tests/data/power_flow/dummy-test-line-into-itself/asym_output_batch.json (hunk 1 above)
Comment (nitbharambe) [reply tests/data/power_flow/dummy-test-line-into-itself/asym_output_batch.json:11]: Given the below issue is solved, these should be fine to modify as the change is within voltage is very low. Power values not modified. (And pf ones can be ignored)

File: tests/data/power_flow/handling-of-links/merge-nodes-by-links-1-line-with-line-to-self/sym_output.json
Comment (nitbharambe) [inline tests/data/power_flow/handling-of-links/merge-nodes-by-links-1-line-with-line-to-self/sym_output.json:66]: The below comment values need to be equal to the link output. from = to.

File: tests/data/short_circuit/dummy-test-line-into-itself/params.json (hunk 2 above)
Comment (Jerry-Jinfeng-Guo) [reply tests/data/short_circuit/dummy-test-line-into-itself/params.json]: Reflected in [74ddc1b](https://github.com/PowerGridModel/power-grid-model/pull/1569/commits/74ddc1bcd98b9c7592efb8b82c12ffd88d644f53)

File: tests/data/short_circuit/dummy-test-line-into-itself/sc_output_batch.json [hunk 4]
```diff
@@ -66,8 +66,8 @@
       ],
       "fault": [
         {"id": 13, "energized": 1, "i_f": [63508331.109258644, 0, 0], "i_f_angle": [-1.4711239498504336, 0, 0]},
-        {"id": 14, "energized": 1, "i_f": [436.49378866976025, 0, 0], "i_f_angle": [2.0624362936211509, 0, 0]},
-        {"id": 15, "energized": 1, "i_f": [0.68295207957823156, 0, 0], "i_f_angle": [-1.5913489370170628, 0, 0]}
+        {"id": 14, "energized": 1, "i_f": [435.89860686970485, 0, 0], "i_f_angle": [2.06320415055763809, 0, 0]},
+        {"id": 15, "energized": 1, "i_f": [435.898606869704856, 0, 0], "i_f_angle": [2.06320415055763828, 0, 0]}
```
Comment (nitbharambe) [inline tests/data/short_circuit/dummy-test-line-into-itself/sc_output_batch.json:70]: Incorrect results because of #1566. These changes incorrect things dont hide any merging of links related issues

**Review by nitbharambe — APPROVED** [review]
